java/src/main/native/src/CudaJni.cpp

```diff
@@ -216,6 +216,20 @@ JNIEXPORT jint JNICALL Java_ai_rapids_cudf_Cuda_getNativeComputeMode(JNIEnv* env
   CATCH_STD(env, -2);
 }
 
+JNIEXPORT jbyteArray JNICALL Java_ai_rapids_cudf_Cuda_getNativeGpuUuid(JNIEnv* env, jclass)
+{
+  try {
+    cudf::jni::auto_set_device(env);
+    int device;
+    CUDF_CUDA_TRY(cudaGetDevice(&device));
+    cudaDeviceProp device_prop;
+    CUDF_CUDA_TRY(cudaGetDeviceProperties(&device_prop, device));
```

Comment:
We already query `cudaGetDeviceProperties` in `Java_ai_rapids_cudf_Cuda_getNativeComputeMode`. Consider refactoring to cache the properties and serve different pieces as needed without going into JNI and CUDA every time. Might also be needed in https://github.com/NVIDIA/spark-rapids/issues/13216

Reply:
This could be a follow-up.

Reply:
It this is true, then the device UUID can be cached. This is a good idea.

Reply:
Follow-up: https://github.com/NVIDIA/spark-rapids/issues/13324